api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionService.java

```diff
@@ -153,6 +164,31 @@ public Subscription createSubscription(final SubscriptionBase subscriptionBase)
         return subscription;
     }
 
+    private Subscription createSubscriptionWithEventTypeLock(final SubscriptionBase subscriptionBase) {
+        try {
+            return eventTypeRepository.lockingTable(EventTypeRepository.TableLock.ROW_EXCLUSIVE, transactionTemplate,
+                    action -> {
+                        final Set<String> dbEventTypes = eventTypeRepository.
+                                listEventTypes(subscriptionBase.getEventTypes()).stream().
+                                map(EventType::getName).collect(Collectors.toSet());
+
+                        if (dbEventTypes.size() != subscriptionBase.getEventTypes().size()) {
```

Comment:
If we start by collecting `missingEventTypes` first we could just check like this:
```java
if (!missingEventTypes.empty()) ...
```